### DISCUSSION TITLE
ML functionality merged!

### DISCUSSION THREAD

**Discussion by kosack** [cta-observatory/ctapipe#2112] [Announcements]
Thanks to many months of work by many, in particular the TU Dortmund group, the (classical) machine-learning branch has finally been merged!  This  is a major step toward version 1.0, and allows us to train and apply reconstruction models for energy regression and particle classification (and in the very near future monoscopic direction reconstruction).   

🎉🍻👏

For users of ctapipe, please check out the development version and do some beta testing to ensure it works as you expect before we release 0.18.0.  

If you encounter any problems, either open an issue, start a discussion here, or chat on the Slack channel.

**Comment by HealthyPear** [comment]
Shouldn't there be an `ml` module at the root of the package or did I misunderstand?
If there should be one, I can't see it from master.

**Reply by maxnoe** [reply]
Due to some issues with circular imports, the machine learning classes are part of `ctapipe.reco`, particularly `ctapipe.reco.sklearn` and not in `ctapipe.ml` anymore.

**Reply by kosack** [reply]
Yes, sorry the title is somewhat wrong: I should have said "ML *functionality* merged", as it is no longer a module! Sorry for the confusion
